### ISSUE TITLE
Expose consensus params from algod

### ISSUE THREAD

**Issue by algorandskiy** [algorand/go-algorand#2148]
Use case:
SDKs need to know some consensus params (examples are max TEAL cost or max ASA URL length.

Suggested implementation:
1. Add REST API endpoint that returns params by protocol name as JSON (or all supported protocols and params)
2. JSON fields must match `ConsensusParams` struct from **config/consensus.go**
3. Add `go:generate` command that dumps all consensus version and params as a single JSON file similarly to TEAL's **langspec.json**

SDKs receive the file from p.3. SDKs are also capable to use actual params by querying the endpoint from p.1

This is similar to TEAL approach: SDKs can work offline using TEAL spec or query the compile endpoint.

**Comment by mxmauro** [comment]
Hi @tsachiherman , I'll handle this and upload a PR

**Comment by tsachiherman** [comment]
@mxmauro - I'm not sure if the above is a good idea anymore.

The ConsensusParams is meant for as an internal data structure for the algod. It's content can change over time, and we don't want to break external dependencies.

I think that the an external component should be using the consensus protocol string directly. The commitment around that are defined in the spec.

Please note that `goal protocols` might give you already what you're looking for.

**Comment by tsachiherman** [comment]
Note that for the SDKs, I think that we should have some static process to update these according to the modified parameters, rather then having the SDK dynamically query the go-algorand repo.

**Comment by mxmauro** [comment]
@tsachiherman I agree but some important parameters might vary in the future breaking apps if the updates are not propagated fast enough.

Two examples:

1. The calculation of a block and TX hash depends on `RequireGenesisHash` added on V15.
2. The minimum fee for a given address partially depends on amounts specified in the consensus.

**Comment by mxmauro** [comment]
`goal protocols` IS what we look for but adding it as an endpoint is very helpful like the `/genesis` endpoint which avoids copying/referencing a file in server environments, mainly, when they are remote.

**Comment by algoanne** [comment]
a use case for why this would be useful: to be able to provide a clear error about why your application failed when submitted. But maybe the solution is to improve our error reporting in general. To be investigated.
